dicoogle/src/main/java/pt/ua/dicoogle/taskManager/RunningIndexTasks.java

```diff
@@ -52,17 +54,38 @@ public static RunningIndexTasks getInstance() {
 	}
 
 	public RunningIndexTasks() {
-		taskRunningList = new HashMap<>();
+		taskRunningList = new LinkedHashMap<>();
 	}
 
 	public void addTask(String taskUid, Task<Report> task) {
 		taskRunningList.put(taskUid, task);
+		if (ENABLE_HOOK){
+            hookRemoveRunningTasks();
+        }
+
 	}
 
 	public boolean removeTask(String taskUid) {
 		return taskRunningList.remove(taskUid) != null;
 	}
 
+
+	public boolean hookRemoveRunningTasks(){
+	    if (this.taskRunningList.size()>SOFT_MAX_RUNNINGTASKS){
+	        int removedTasks = 0 ;
+	        Iterator<String> iterator = this.taskRunningList.keySet().iterator();
+	        while(iterator.hasNext()&& removedTasks<NUMBER_RUNNINGTASKS_TO_CLEAN){
+	            String tId = iterator.next();
+                Task t =  this.taskRunningList.get(tId);
+                if (t.isCancelled() || t.isDone()){
+                    iterator.remove();
+                    removedTasks++;
+                }
+            }
+        }
+	    return true;
```

Comment:
The method always returns `true`. It should either provide additional information through the returned value (e.g. whether at least one task was removed) or just return `void`.

Reply:
Thanks for feedback @Enet4. I will address the minor issues. The code at `dev` branch is already formated by the plugin already, so it will be easier to merge.

Regarding the other issues, sure, we can kept them for newest versions of Dicoogle 3 or 4 . For now, we need to close this one, close the 3.0.0 and move forward :)

Reply:
> The code at dev branch is already formated by the plugin already, so it will be easier to merge.

Ah right, the 2.X development branch does not have standardized code formatting. We can look into that later if it becomes a nuisance. :)

I say it's good to merge once the other minor concerns are resolved.